File: POOUber/java/Car.java
class Car {
    private Integer id;
    private String license;
    private Account driver;
    protected Integer passenger;

    public Car(String license, Account driver){
        this.license = license;
        this.driver = driver;
/*         passenger = 3;
        System.out.println("Passenger: " + passenger); */
    }

    void printDataCar(){
        if(passenger != null){
            System.out.println(

                "El nombre del conductor es " + driver.name + " y su licencia es " + license
            );
        }
    }

    //Uso de getter and setter
    public Integer getPassenger(){
        return passenger;
    }

    public void setPassenger(Integer passenger){
        if(passenger == 4){
        this.passenger = passenger;
        } else{
            System.out.println("El valor debe ser 4");
        }
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getLicense() {
        return license;
    }

    public void setLicense(String license) {
        this.license = license;
    }

    public Account getDriver() {
        return driver;
    }

    public void setDriver(Account driver) {
        this.driver = driver;
    }


}

File: POOUber/js/Car.js
/* function Car(license, driver){
    this.id;
    this.license = license;
    this.driver = driver;
    this.passenger;
}

Car.prototype.printDataCar = function(){
    console.log(this.driver);
    console.log(this.driver.name);
    console.log(this.driver.document);
} */

class Car{
    constructor(license, driver){
        this.id;
        this.license = license;
        this.driver = driver;
        this._passenger;
    }

    get passenger(){
        return this._passenger;
    }

    set passenger(passengers){
        if (passengers == 4){
            this._passenger = passengers;
        } else{
            console.log("Debes tener 4 pasajeros");
        }
    }

    printDataCar(){
        console.log(this.driver);
        console.log('Driver Name: ' + this.driver.name);
        console.log('Driver Document: ' + this.driver.document);
        console.log('Passengers: ' + this.passenger);
    }
}